pkg/sink/sink_test.go

```diff
@@ -1192,16 +1207,20 @@ func TestExecuteInterceptor_ExtensionChaining(t *testing.T) {
 	if err != nil {
 		t.Fatalf("http.NewRequest: %v", err)
 	}
-	body := fmt.Sprintf(`{"sha": "%s"}`, sha)
+	body := fmt.Sprintf(`{"pull_request_url": "%s"}`, pull_request_url)
 	resp, _, iresp, err := s.ExecuteInterceptors(trigger, req, []byte(body), logger, eventID)
 	if err != nil {
 		t.Fatalf("executeInterceptors: %v", err)
 	}
 
 	wantBody := map[string]interface{}{
-		"sha": sha,
+		"pull_request_url": pull_request_url,
 		"extensions": map[string]interface{}{
-			"truncated_sha": "abcde",
+			"add_team_members": "something",
+			"add_pr_body": map[string]interface{}{
+				"pull_request_body": "key",
```

Comment:
When we run this test it fails because the `pull_request_body` key is gone. When the second `Webhook` is executed, **the `add_pr_body` key added by first cel interceptor in extensions gets re-applied so the keys added to `add_pr_body` on the first `Webhook` are lost when it reaches the second `Webhook`.**

Reply:
yeah I think I get it....this happens if we are using nested json keys i.e. if both keys were added at the top-level instead of nested under `add_pr_body` I think this behavior would not happen.


I think the lines to blame is here: https://github.com/tektoncd/triggers/blob/master/pkg/sink/sink.go#L351

We are simply accounting for top level keys (like  add_pr_body) and resetting its value to the one from request.Extensions. Instead we need to do some sort of a merge. 

One way might be to marshal the body into a map[string]interface{}, then do a json merge, and then turn it back into a []byte.